### ISSUE TITLE
Ability to start timelapse without printing

### ISSUE THREAD

**Issue by frankhommers** [OctoPrint/OctoPrint#452]
Sometimes I use the (hardware) control panel of my printer to start a job, I would really like the ability to make a timelapse of that (e.g. not bound to the print start/done/etc. events).

**Comment by callimero** [comment]
+1 :+1:  Also nice for testing and making stop motion movies on thebuild platform (haha :-)

**Comment by Diaoul** [comment]
+1

**Comment by AquaticsLive** [comment]
Really want this as well.  I hardware print from SD card on my really long prints for safety.  I would love to use Octoprint to records webcam for those.  I use Yawcam for now, just put the IP of the camera connection copied from Octoprint settings and bingo it records.

**Comment by foosel** [comment]
> I hardware print from SD card

Just for the record, you know you can start those prints from SD card through OctoPrint as well? It will still do the same as if you started it through the printers controller, only also give you progress feedback and also timelapse functionality, just like for prints that are streamed over USB...

**Comment by AquaticsLive** [comment]
I know this has been a long time, but gave printing from SD in Octoprint a try for a 37 hour print this weekend.

When the Raspberry PI froze the printer stopped printing I fixed the print so no big loss, but wanted to post my results here of the test.

**Comment by frankhommers** [comment]
> Just for the record, you know you can start those prints from SD card through OctoPrint as well? It will still do the same as if you started it through the printers controller, only also give you progress feedback and also timelapse functionality, just like for prints that are streamed over USB...

@foosel: Thats not exactly true. When I change the material and enter this through the hardware control panel on my Ultimaker 2, the start up sequence differs when I use the panel for printing. Therefore I would still love this feature.

**Comment by chriscohoat** [comment]
I adapted some of the internal methods from `src/octoprint/timelapse.py` as a simple way to manually save timelapse images. It can be found here: https://github.com/chriscohoat/octoprint-scripts/blob/master/src/manual_timelapse/main.py

My use case is using the webcam I have setup with Octopi to take a timelapse of the construction of my new printer.

@foosel my implementation is obviously basic so far, but if you wanted help extending the web interface to manually create timelapses I'd be happy to take a crack at it. There are a few different features outstanding for that I believe.

**Comment by foosel** [comment]
@chriscohoat yes there are. Not only for manual timelapse starting/stopping though but also for other timelapse modes and adjusted starting and stopping points, which is why I'm currently considering this as part of a larger timelapse implementation refactoring, allowing additional timelapse types and different starting/stopping behaviour through plugins.

So that would make it a bit larger than what you currently have in mind I guess ;)

I'd still be open to a PR just for manual start/stop for now though - that would then have to be later adapted to the more modular approach, but that shouldn't be a problem. I have to admit though that I'm currently drawing a complete blank with regards to how to do something like that from a strict UI/UX point of view - if you have an idea, maybe some sketches in a "[Brainstorming]" ticket might be a good idea to get the discussion going :)

**Comment by chriscohoat** [comment]
@foosel larger timelapse refactoring sounds like more fun :D!

Brainstorming ticket sounds like the way to go. If there are any specific tickets you have in mind let me know...otherwise I'll glean through the issues that are open/closed that mention timelapse functionality. I'll at least be on the lookout for different modes and starting/stopping points, as well as anything else that looks useful. 

Look for that in the next few days. I'm still waiting on my printer to show up ;)

My first instinct is to move the "Timelapse Configuration" into settings (with maybe a smaller direct link to get to it in the Timelapse tab) which could leave more space for some of the new features. I also wonder if it's worth linking the video stream in the Timelapse tab just like the Control tab already does...as a new user anyway I expected it to be there. It would make the UX/controls more intuitively possibly (depending on what gets accepted as a feature, I guess).

**Comment by JaredFarnum** [comment]
Any news on this? I would love to be able to record a timelapse without using OctoPrint to print. I use Octoprint for remote monitoring only right now.

**Comment by mabnz** [comment]
It'd be great to be able to do this.

**Comment by CyrisXD** [comment]
Any news on this? Using Octopi to solely monitor prints from my phone, would be awesome to manually start/stop time lapses as well.

**Comment by bernd87** [comment]
Please implement manuel start for interval timelapse

**Comment by Docaed** [comment]
It seems that this cannot be accomplished as it was first mentioned in 2014 and it is now 2019.  Can any body answer that question?  It seems like a valuable option but not obtainable.  Can any one suggest another software for taking time lapse videos of a print job.  I too have issues using my new PRUSA MK3s printing through optoprint, so I just print from the SD card with no issues.  I like that I can view the print job on octoprint, but it would be nice to have an option to time lapse off line as well.

**Comment by foosel** [comment]
> It seems that this cannot be accomplished as it was first mentioned in 2014 and it is now 2019

There are more pressing issues and feature requests than turning OctoPrint into a standalone timelapse solution.

> it would be nice to have an option to time lapse off line as well

https://www.google.com/search?q=timelapse+raspberry+pi

**Comment by frankhommers** [comment]
> > It seems that this cannot be accomplished as it was first mentioned in 2014 and it is now 2019
> 
> There are more pressing issues and feature requests than turning OctoPrint into a standalone timelapse solution.

It's not standalone timelapse. I still use everything else from OctoPrint. Especially monitoring everything while printing (initiated by something else than OctoPrint).

**Comment by Steve5323** [comment]
Well, it would be nice to have since I'm trying to record an SLA printer that only prints from usb.  No other time lapse programs work as well for me as octoprint.  I guess I'm sol

**Comment by RobertoC27** [comment]
would love to see this feature being implemented, how can i help?

**Comment by LazeMSS** [comment]
This is now possible in:
https://plugins.octoprint.org/plugins/octolapse/

**Comment by Letme** [comment]
Voting for implementation of this in 2021. I can also do some digging since it seems nobody else started.

**Comment by cp2004** [comment]
Why does this have to be implemented in OctoPrint? Surely you could just use a 3rd party timelapse software? There's nothing special about the timelapse in OctoPrint *other than automatically starting*. Even if you need to call a command line to start it, you could put that into the system command menu.

**Comment by Letme** [comment]
You could do of course, but since OctoPrint is already installed and running I assume starting a timelapse with a button instead of job trigger is a minor tweak. Even if OctoPrint is not in control of the job, I still use features like emergency stop from it. I tried with virtual printer, but job there completed much faster than real one. We could stop timelapse once nozzle temperature drops below 50 for example.

**Comment by Letme** [comment]
Or even better, you can stop timelapse on `Recv: Not SD printing` (hopefully that is general response from idle printer?)

**Comment by ESurge** [comment]
Use the virtual printer to start a long print. This will start creating the timelapse. When you want to stop, cancel the print to render the timelapse. While not as elegant as an actual manual start/stop function, it's a workaround.
